arch/risc-v/src/common/riscv_fpucmp.c

```diff
@@ -37,58 +34,27 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
-/* Configuration ************************************************************/
-
-#undef HAVE_FPU
-#if defined(CONFIG_ARCH_FPU) && \
-    !defined(CONFIG_TESTING_OSTEST_FPUTESTDISABLE) && \
-    defined(CONFIG_TESTING_OSTEST_FPUSIZE) && \
-    defined(CONFIG_SCHED_WAITPID)
-#    define HAVE_FPU 1
-#endif
-
-#ifdef HAVE_FPU
-
-#if CONFIG_TESTING_OSTEST_FPUSIZE != (4 * FPU_XCPT_REGS)
-#  error "CONFIG_TESTING_OSTEST_FPUSIZE has the wrong size"
-#endif
+#ifdef CONFIG_ARCH_FPU
 
 /****************************************************************************
- * Private Data
+ * Public Functions
  ****************************************************************************/
 
-static uint32_t g_saveregs[XCPTCONTEXT_REGS];
-
 /****************************************************************************
- * Public Functions
+ * Name: up_fpucmp
+ *
+ * Description:
+ *   compare FPU areas from thread context
+ *
  ****************************************************************************/
 
-/* Given an array of size CONFIG_TESTING_OSTEST_FPUSIZE, this function will
- * return the current FPU registers.
- */
-
-void arch_getfpu(FAR uint32_t *fpusave)
+bool up_fpucmp(FAR const void *saveregs1, FAR const void *saveregs2)
 {
-  irqstate_t flags;
-
-  /* Take a snapshot of the thread context right now */
-
-  flags = enter_critical_section();
-  riscv_saveusercontext(g_saveregs);
+  uintptr_t regs1 = (uintptr_t)saveregs1;
+  uintptr_t regs2 = (uintptr_t)saveregs2;
 
-  /* Return only the floating register values */
-
-  memcpy(fpusave, &g_saveregs[INT_XCPT_REGS], (4*FPU_XCPT_REGS));
-  leave_critical_section(flags);
+  return memcmp((void *)(regs1 + INT_REG_SIZE * INT_XCPT_REGS),
```

Comment:
Maybe next can be considered as an option
```
  uintptr_t *regs1 = (uintptr_t *)saveregs1;
  uintptr_t *regs2 = (uintptr_t *)saveregs2;

  return memcmp(regs1 + INT_XCPT_REGS),
       regs2 + INT_XCPT_REGS),
       sizeof(uintptr_t) * FPU_XCPT_REGS) == 0;
```

Reply:
Done